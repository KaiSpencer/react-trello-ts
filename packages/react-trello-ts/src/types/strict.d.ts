declare module "uuid/v1" {
	export default function uuid(): string;
}
