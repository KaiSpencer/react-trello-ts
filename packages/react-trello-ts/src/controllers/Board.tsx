import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
