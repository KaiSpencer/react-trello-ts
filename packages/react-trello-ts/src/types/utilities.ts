/**
 * Given a type `TType` and a type `TRequired` which is a subset of required keys of `TType`, return a new type which is the
 * same as `TType` but with all keys optional except those in `TRequired`.
 */
export type PartialExcept<
	TType,
	TRequired extends keyof TType,
> = Partial<TType> & Pick<TType, TRequired>;
