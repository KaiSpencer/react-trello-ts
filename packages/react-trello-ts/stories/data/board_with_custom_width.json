{
  "lanes": [
    {
      "id": "PLANNED",
      "title": "Planned Tasks",
      "label": "20/70",
      "style": {"width": 280},
      "cards": [
        {
          "id": "Milk",
          "title": "Buy milk",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "2 Gallons of milk at the Deli store"
        },
        {
          "id": "Plan2",
          "title": "Dispose Garbage",
          "label": "10 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Sort out recyclable and waste as needed"
        },
        {
          "id": "Plan3",
          "title": "Write Blog",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Can AI make memes?"
        },
        {
          "id": "Plan4",
          "title": "Pay Rent",
          "label": "5 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Transfer to bank account"
        }
      ]
    },
    {
      "id": "WIP",
      "title": "Work In Progress",
      "label": "10/20",
      "style": {"width": 280},
      "cards": [
        {
          "id": "Wip1",
          "title": "Clean House",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses"
        }
      ]
    },
    {
      "id": "BLOCKED",
      "title": "Blocked",
      "label": "0/0",
      "style": {"width": 280},
      "cards": []
    },
    {
      "id": "COMPLETED",
      "title": "Completed",
      "style": {"width": 280},
      "label": "2/5",
      "cards": [
        {
          "id": "Completed1",
          "title": "Practice Meditation",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Use Headspace app"
        },
        {
          "id": "Completed2",
          "title": "Maintain Daily Journal",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Use Spreadsheet for now"
        }
      ]
    },
    {
      "id": "REPEAT",
      "title": "Repeat",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Repeat1",
          "title": "Morning Jog",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Track using fitbit"
        }
      ]
    },
    {
      "id": "ARCHIVED",
      "title": "Archived",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    },
    {
      "id": "ARCHIVED2",
      "title": "Archived2",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    },
    {
      "id": "ARCHIVED3",
      "title": "Archived3",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    }
  ]
}
