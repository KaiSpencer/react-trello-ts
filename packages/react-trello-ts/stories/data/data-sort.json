
{
  "lanes": [
    {
      "id": "SORTED_LANE",
      "title": "Sorted Lane",
      "label": "20/70",
      "cards": [
        {
          "id": "Card1",
          "title": "Buy milk",
          "label": "2017-12-01",
          "description": "2 Gallons of milk at the Deli store",
          "metadata": {
            "completedAt": "2017-12-01T10:00:00Z",
            "shortCode": "abc"
          }
        },
        {
          "id": "Card2",
          "title": "Dispose Garbage",
          "label": "2017-11-01",
          "description": "Sort out recyclable and waste as needed",
          "metadata": {
            "completedAt": "2017-11-01T10:00:00Z",
            "shortCode": "aaa"
          }
        },
        {
          "id": "Card3",
          "title": "Write Blog",
          "label": "2017-10-01",
          "description": "Can AI make memes?",
          "metadata": {
            "completedAt": "2017-10-01T10:00:00Z",
            "shortCode": "fa1"
          }
        },
        {
          "id": "Card4",
          "title": "Pay Rent",
          "label": "2017-09-01",
          "description": "Transfer to bank account",
          "metadata": {
            "completedAt": "2017-09-01T10:00:00Z",
            "shortCode": "ga2"
          }
        }
      ]
    }
  ]
}
