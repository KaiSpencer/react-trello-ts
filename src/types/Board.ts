import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
