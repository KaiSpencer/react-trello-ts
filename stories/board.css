.boardContainer {
    background-color: #4BBF6B;
}