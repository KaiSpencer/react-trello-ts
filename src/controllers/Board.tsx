import classNames from "classnames";
import React, { FC, PropsWithChildren, createContext } from "react";
import { v1 } from "uuid";
import * as DefaultComponets from "../components";
import { store } from "../store/store";
import { BoardData } from "../types/Board";
import { BoardContainer } from "./BoardContainer";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			{/*
			// @ts-ignore */}
			<components.GlobalStyle />
			<BoardContainer
				id={id || v1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
