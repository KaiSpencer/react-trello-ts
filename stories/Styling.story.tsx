import { Meta, StoryObj } from "@storybook/react";
import React from "react";

import Board from "../src";
import { BoardData } from "../src/types/Board";
import "./board.css";
import data from "./data/base.json";
Board.displayName = "Board";

const dataWithLaneStyles = {
	lanes: [
		{
			id: "PLANNED",
			title: "Planned Tasks",
			label: "20/70",
			style: {
				width: 280,
				backgroundColor: "#3179ba",
				color: "#fff",
				boxShadow: "2px 2px 4px 0px rgba(0,0,0,0.75)",
			},
			cards: [
				{
					id: "Milk",
					title: "Buy milk",
					label: "15 mins",
					description: "2 Gallons of milk at the Deli store",
				},
				{
					id: "Plan2",
					title: "Dispose Garbage",
					label: "10 mins",
					description: "Sort out recyclable and waste as needed",
				},
			],
		},
		{
			id: "DONE",
			title: "Doned tasks",
			label: "10/70",
			style: {
				width: 280,
				backgroundColor: "#ba7931",
				color: "#fff",
				boxShadow: "2px 2px 4px 0px rgba(0,0,0,0.75)",
			},
			cards: [
				{
					id: "burn",
					title: "Burn Garbage",
					label: "10 mins",
					description: "Sort out recyclable and waste as needed",
				},
			],
		},
		{
			id: "ARCHIVE",
			title: "Archived tasks",
			label: "1/2",
			cards: [
				{
					id: "archived",
					title: "Archived",
					label: "10 mins",
				},
			],
		},
	],
};

const dataWithCardStyles: BoardData = {
	lanes: [
		{
			id: "PLANNED",
			title: "Planned Tasks",
			label: "20/70",
			cards: [
				{
					id: "Milk",
					title: "Buy milk",
					label: "15 mins",
					description: "2 Gallons of milk at the Deli store",
					style: { backgroundColor: "#eec" },
				},
				{
					id: "Plan2",
					title: "Dispose Garbage",
					label: "10 mins",
					description: "Sort out recyclable and waste as needed",
				},
				{
					id: "Plan3",
					title: "Burn Garbage",
					label: "20 mins",
				},
			],
		},
	],
};

export default ({
	title: "Styling",
	component: Board,
} satisfies Meta<typeof Board>);

type Story = StoryObj<typeof Board>;

export const BoardStyling: Story = {
	name: "Board Styling",
	render: (args) => (
		<Board
			data={data}
			style={{ padding: "30px 20px", fontFamily: "Verdana" }}
			className="boardContainer"
		/>
	),
};

export const LaneStyling: Story = {
	name: "Lane Styling",
	render: (args) => (
		<Board
			data={dataWithLaneStyles}
			laneStyle={{ backgroundColor: "#666" }}
			style={{ backgroundColor: "#eee" }}
		/>
	),
};

export const CardStyling: Story = {
	name: "Card Styling",
	render: (args) => (
		<Board data={dataWithCardStyles} cardStyle={{ backgroundColor: "#ffe" }} />
	),
};
