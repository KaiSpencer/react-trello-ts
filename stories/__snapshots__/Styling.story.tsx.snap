// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styling BoardStyling smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
     style="padding: 30px 20px; font-family: Verdana;"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Planned Tasks
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              20/70
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Milk"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Buy milk
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    15 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  2 Gallons of milk at the Deli store
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Dispose Garbage
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    10 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Sort out recyclable and waste as needed
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan3"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Write Blog
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    30 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Can AI make memes?
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan4"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Pay Rent
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    5 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Transfer to bank account
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Work In Progress
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              10/20
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Wip1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Clean House
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    30 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Blocked
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              0/0
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Completed
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              2/5
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Completed1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Practice Meditation
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    15 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Use Headspace app
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Completed2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Maintain Daily Journal
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    15 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Use Spreadsheet for now
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Repeat
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/1
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Repeat1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Morning Jog
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    30 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Track using fitbit
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Archived
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/1
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Archived1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Go Trekking
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    300 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Completed 10km on cycle
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Archived2
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/1
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Archived2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Go Jogging
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    300 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Completed 10km on cycle
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Archived3
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/1
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Archived3"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Go Cycling
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    300 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Completed 10km on cycle
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`Styling CardStyling smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Planned Tasks
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              20/70
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Milk"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                       style="background-color: rgb(238, 238, 204);"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Buy milk
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    15 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  2 Gallons of milk at the Deli store
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                       style="background-color: rgb(255, 255, 238);"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Dispose Garbage
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    10 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Sort out recyclable and waste as needed
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan3"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                       style="background-color: rgb(255, 255, 238);"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Burn Garbage
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    20 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`Styling LaneStyling smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
     style="background-color: rgb(238, 238, 238);"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px; background-color: rgb(49, 121, 186); color: rgb(255, 255, 255); box-shadow: rgba(0, 0, 0, 0.75) 2px 2px 4px 0px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Planned Tasks
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              20/70
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Milk"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Buy milk
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    15 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  2 Gallons of milk at the Deli store
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Plan2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Dispose Garbage
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    10 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Sort out recyclable and waste as needed
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="width: 280px; background-color: rgb(186, 121, 49); color: rgb(255, 255, 255); box-shadow: rgba(0, 0, 0, 0.75) 2px 2px 4px 0px;"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Doned tasks
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              10/70
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="burn"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Burn Garbage
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    10 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  Sort out recyclable and waste as needed
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
               style="background-color: rgb(102, 102, 102);"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Archived tasks
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              1/2
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="archived"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    Archived
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    10 mins
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
